src/components/Form.js

```diff
@@ -1,12 +1,28 @@
-const Form = () => (
-  <div className="form">
-    <form>
-      <h3>Add a Book</h3>
-      <input type="text" name="title" placeholder="title" />
-      <input type="text" name="author" placeholder="author" />
-      <button type="submit">Add Book</button>
-    </form>
-  </div>
-);
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/booksSlice';
 
+const Form = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+
+  const addBookHandler = (e) => {
+    e.preventDefault();
+    dispatch(addBook({ id: '$(Math.random() * 100)', title, author }));
```

Comment:
- [ ] While dispatching `addbook` the id is created as a string `'$(Math.random() * 100)'` and that's not ok because the id suppose to be unique and this string is passed every time you create a new book, Please fix that.

![image](https://user-images.githubusercontent.com/68796232/225081671-1ad21c02-e736-45d3-9fcc-ce53dedf565f.png)

[hint] You can do it like that:
```javascript
dispatch(addBook({ id: Math.floor(Math.random() * 1000), title, author }));
```

Reply:
Thanks for the tip.